src/main/java/Storage.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Storage class serve as a storage of all tasks and execute command
+ * on the list of tasks
+ */
+public class Storage {
+    private static final int TODO_LENGTH = 5;
+    private static final int DEADLINE_LENGTH = 9;
+    private static final int EVENT_LENGTH = 6;
+    private static final String COMMAND_SEPARATOR = " ";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_MARK = "mark";
+    private static final String COMMAND_UNMARK = "unmark";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_NULL = "";
+    private static final int MAXIMUM_LIST_SIZE = 120;
+    private static final Task[] list = new Task[MAXIMUM_LIST_SIZE];
+    private static int size = 0;
+
+    /**
+     * Add item to list of tasks
+     *
+     * @param item tasks in the list
+     */
+    public static void add(Task item) {
+        list[size] = item;
+        size++;
+        System.out.format("added: %s%n%d items on list%n", item, size);
+    }
+
+    /**
+     * List all items in storage
+     */
+    public static void listAll() {
+        for (int i = 0; i < size; i++) {
+            System.out.format("%d.%s%n", i + 1, list[i]);
+        }
+    }
+
+    /**
+     * toggleMarkStatus changes the check mark of each item in storage
+     *
+     * @param cmd COMMAND_MARK or COMMAND_UNMARK
+     */
+    public static void toggleMarkStatus(String cmd) {
+        String[] cmds = cmd.split(COMMAND_SEPARATOR);
+
+        try {
+            if (cmds[0].equalsIgnoreCase(COMMAND_MARK)) {
+                list[Integer.parseInt(cmds[1]) - 1].setIsDone(true);
```

Comment:
Consider breaking this up to avoid complicated expressions.

https://nus-cs2113-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions